exec/java-exec/src/main/java/org/apache/drill/exec/physical/base/ScanStats.java

```diff
@@ -23,27 +23,27 @@
 
   public static final ScanStats ZERO_RECORD_TABLE = new ScanStats(GroupScanProperty.EXACT_ROW_COUNT, 0, 1, 1);
 
-  private final long recordCount;
-  private final float cpuCost;
-  private final float diskCost;
+  private final double recordCount;
+  private final double cpuCost;
+  private final double diskCost;
   private final GroupScanProperty property;
 
-  public ScanStats(GroupScanProperty property, long recordCount, float cpuCost, float diskCost) {
+  public ScanStats(GroupScanProperty property, double recordCount, double cpuCost, double diskCost) {
```

Comment:
There are other GroupScanPop where the rowCount stats for example is converted from double to long while creating ScanStats. Whereas in most of the cases rowCount is long only. 
Since we are making this change for treating rowCount or other stats as double, shouldn't we update other GroupScanPop's as well to make sure they are creating stats using correct type ?

[Example of rowCount with double type but converted to long.](https://github.com/apache/drill/blob/master/contrib/storage-jdbc/src/main/java/org/apache/drill/exec/store/jdbc/JdbcGroupScan.java#L82)
[Example of rowCount with long type.](https://github.com/apache/drill/blob/master/contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/JsonTableGroupScan.java#L188)

Reply:
Yes, you are correct. There are more examples of where we should do so e.g. HBaseGroupScan. However, the original JIRA was to fix the integer overflow so I skipped these changes. I will create another JIRA to address these issues.

Reply:
LGTM +1